src/mono/mono/mini/intrinsics.c

```diff
@@ -556,7 +556,12 @@ MONO_RESTORE_WARNING
 		t = mini_get_underlying_type (t);
 		if (cfg->gshared && t != ctx->method_inst->type_argv [0] && MONO_TYPE_ISSTRUCT (t) && mono_class_check_context_used (mono_class_from_mono_type_internal (t)))
 			cfg->prefer_instances = TRUE;
-		return mini_emit_memory_load (cfg, t, args [0], 0, MONO_INST_UNALIGNED);
+
+		#ifdef NO_UNALIGNED_ACCESS
+			return mini_emit_memory_load (cfg, t, args [0], 0, MONO_INST_UNALIGNED);
+		#else
+			return mini_emit_memory_load (cfg, t, args [0], 0, 0);
+		#endif
```

Comment:
Doesn't it make more sense to have the check inside of `mini_emit_memory_load`?

Reply:
I wasn't 100% confident that this solution will be the correct one so for now I put it only for `ReadUnaligned` as it was the immediate solution for the regression. I agree that it would be better to have it inside `mini_emit_memory_load` to optimize all memory load emits. I could move the `#ifdef` to `mini_emit_memory_load` method and zero-out the `MONO_INST_UNALIGNED` flag when `NO_UNALIGNED_ACCESS` is not set. What do you think?

Reply:
I think it should be enough to just if-out the if there. Also, `mini_emit_memory_store` should probably be changed too.